### PR TITLE
feat: invoice pay service

### PR DESCRIPTION
**Pull request by KAVAN-DESAI**

### CODE REVIEW

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
Good starting point, I added a couple of comments

File: lib/src/cln_grpc_base.dart [hunk 1]
```diff
@@ -132,6 +133,18 @@ class GRPCClient extends LightningClient {
     return response as T;
   }
 
+  Future<T> pay<T>(
+      {required PayRequest params, T Function(Map)? onDecode}) async {
+    /// request to server
+    var response = await stub.pay(params);
+    print("inside");
+    print(response);
```
[the remark this answers is not in the document]
Comment (KAVAN-DESAI) [reply lib/src/cln_grpc_base.dart:141]: ooh shit sorry my bad

File: lib/src/cln_grpc_base.dart [hunk 2]
```diff
@@ -164,6 +177,14 @@ class GRPCClient extends LightningClient {
       case "listinvoices":
         return await listInvoices(
             params: params.as<ListinvoicesRequest>(), onDecode: onDecode);
+      case "pay":
+        Amount amount = Amount();
+        amount.msat =
+            Int64(int.parse(params.toJSON()["msatoshi"]["msat"].toString()));
+        return await pay(
+            params: PayRequest(
+                bolt11: params.toJSON()["bolt11"].toString(), msatoshi: amount),
```
[the remark this answers is not in the document]
Comment (KAVAN-DESAI) [reply lib/src/cln_grpc_base.dart:186]: Yeah should work!!

File: example/cln_grpc_example.dart [hunk 3]
```diff
@@ -51,14 +52,14 @@ Future<void> main(List<String> args) async {
...
-  // Int64 x=Int64(10000);
-  // Amount amount=Amount();
-  // amount.msat=x;
-  // String boltString="";
-  // var pay  = await client.call<PayProxy, PayResponse>(
-  //     method: "pay", params: PayProxy.build(boltString,amount));
-  //
-  // print(pay);
+  /// custom call using proxy to add dynamic inputs
+  Int64 x = Int64(1000);
```
Comment (KAVAN-DESAI) [inline example/cln_grpc_example.dart:56]: `Int64 msat = Int64(1000);`

**Review by vincenzopalazzo — COMMENTED** [review]
We can put the code example also inside the tests? to make sure that this works? :)

It is too boring clone the repo only to test that this feature works 😄

File: example/cln_grpc_example.dart [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import 'package:cln_common/cln_common.dart';
 import 'package:cln_grpc/cln_grpc.dart';
+import 'package:fixnum/fixnum.dart';
```
Comment (vincenzopalazzo) [inline example/cln_grpc_example.dart:3]: we really need this?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: example/cln_grpc_example.dart (hunk 4 above)
Comment (KAVAN-DESAI) [reply example/cln_grpc_example.dart:3]: Yes, We need to declare msatoshi with the type `Amount`, so to convert integer to `Int64` we need it

Here ```Int64 msat = Int64(1000);```

Comment (vincenzopalazzo) [reply example/cln_grpc_example.dart:3]: Cool! more deps hahah

**Review by vincenzopalazzo — APPROVED** [review]
Squashing the commit and ready to land.

Good job! and with the test we found a bug, and this is cool!

**Pull request merged**